pkg/rpc/client/helper.go

```diff
@@ -97,33 +106,107 @@ func topMapFromStack(st []stackitem.Item) (*stackitem.Map, error) {
 	return st[index].(*stackitem.Map), nil
 }
 
-// topIterableFromStack returns top list of elements of `resultItemType` type from the stack.
-func topIterableFromStack(st []stackitem.Item, resultItemType interface{}) ([]interface{}, error) {
-	index := len(st) - 1 // top stack element is last in the array
-	if t := st[index].Type(); t != stackitem.InteropT {
-		return nil, fmt.Errorf("invalid return stackitem type: %s (InteropInterface expected)", t.String())
+func (c *Client) invokeAndPackIteratorResults(contract util.Uint160, operation string, params []smartcontract.Parameter, signers []transaction.Signer) (*result.Invoke, error) {
+	// We can't use TraverseIterator for it, because this functionality is disabled
+	// on the mainnet/testnet, see https://github.com/neo-project/neo-modules/pull/715#discussion_r897641254
+	// Thus, we'll create a short program that traverses iterator got from System.Contract.Call,
+	// packs all its values into array and stores it on stack.
+	script := io.NewBufBinWriter()
+	emit.Instruction(script.BinWriter, opcode.INITSLOT, // Initialize local slot...
+		[]byte{
+			2, // with 2 local variables (0-th for iterator, 1-th for the resulting array)...
+			0, // and 0 arguments.
+		})
+	// Pack arguments for System.Contract.Call.
+	if len(params) == 0 {
+		emit.Opcodes(script.BinWriter, opcode.NEWARRAY0)
+	} else {
+		bytes, err := json.Marshal(params)
+		if err != nil {
+			return nil, fmt.Errorf("failed to convert operation parameters into System.Contract.Call arguments: marshalling error %w", err)
+		}
+		var slice request.Params
+		err = json.Unmarshal(bytes, &slice)
```

Comment:
Actually, this marshal/unmarshal code performs exactly the same thing that client/server perform during `invokefunction` call processing. The case is that `ExpandArrayIntoScript` works with `request.Param` only and has such artifacts like emitting `Boolean`s like `Int`s:
```
case smartcontract.BoolType:
	val, err := fp.Value.GetBoolean() // not GetBooleanStrict(), because that's the way C# code works
	if err != nil {
		return errors.New("not a bool")
	}
	if val {
		emit.Int(script, 1)
	} else {
		emit.Int(script, 0)
	}
```
To remove marshal/unmarshal we have to create a copy of `ExpandArrayIntoScript` for `smartcontract.Parameter`, which will be almost exact copy of `ExpandArrayIntoScript`. I'm not sure we really need this.

Reply:
So do you think it's better to create an analogue of `ExpandArrayIntoScript` for `smartcontrct.Parameter`?

Reply:
Fixed by creating a separate function for `smartcontract.Parameter`.